src/renameAll.js

```diff
@@ -0,0 +1,13 @@
+const apply   = require('ramda/src/apply')
+const compose = require('ramda/src/compose')
+const curry   = require('ramda/src/curry')
+const map     = require('ramda/src/map')
+const toPairs = require('ramda/src/toPairs')
+
+const rename = require('./rename')
+
+// renameAll :: { k: v } -> { k: v } -> { k: v }
+const renameAll = (renames, obj) =>
+  compose(apply(compose), map(apply(rename)), toPairs)(renames)(obj)
```

Comment:
`apply(compose)` will error if `obj` only has <= 1 value, yes?

Reply:
```
> compose(Function.prototype)
[Function]
> compose()
Error: compose requires at least one argument
```
So it looks like `renameAll({})` wouldn't work, sure.  But you'd get such a nice descriptive error (see above) to say what you did wrong.

Reply:
you should be able to pass `{}` without error, imagine if that object is dynamically constructed, you wouldn't want it to error in that case.

Reply:
Sorry, I should have added the :trollface:.

Reply:
lol, you fooled me